### ISSUE TITLE
Images get corrupted when performing multiple modifications

### ISSUE THREAD

**Issue by Netmosfera** [Imagick/imagick#503]
```
<?php
// doesn't work if you apply filters one after another

$imagick = new Imagick(__DIR__ . "/use_any_photo.png");

$mask = new Imagick();
$mask->newImage(2560, 1440, "white");
$drawing = new ImagickDraw();
$drawing->setBorderColor('black');
$drawing->setFillColor('black');
$drawing->ellipse(2560/2, 1440/2, 2560/2, 1440/2, 0, 360);
$mask->drawImage($drawing);
$imagick->setImageMask($mask, Imagick::PIXELMASK_WRITE);
$imagick->brightnessContrastImage(30+40, 36+40);

$imagick->unsharpMaskImage(1, 0.5, 1, 0);

$imagick->writeImage(__DIR__ . "/result1.png");
```

```
<?php

// works if you save on a temporary image every time right after applying a filter

$imagick = new Imagick(__DIR__ . "/use_any_photo.png");

$mask = new Imagick();
$mask->newImage(2560, 1440, "white");
$drawing = new ImagickDraw();
$drawing->setBorderColor('black');
$drawing->setFillColor('black');
$drawing->ellipse(2560/2, 1440/2, 2560/2, 1440/2, 0, 360);
$mask->drawImage($drawing);
$imagick->setImageMask($mask, Imagick::PIXELMASK_WRITE);
$imagick->brightnessContrastImage(30+40, 36+40);
$imagick->setImageFormat("png");
$temp = $imagick->getImageBlob();

$imagick = new Imagick();
$imagick->readImageBlob($temp);
$imagick->unsharpMaskImage(1, 0.5, 1, 0);

$imagick->writeImage(__DIR__ . "/result2.png");
```

This might not be a bug, just some strange default maybe, because changing Imagick::CHANNEL_ seems to make it worse/better.

But if it is a bug, it doesn't affect just sharpen and brightness/contrast. As far I can tell, you always get some amount of artifacts. Sometimes they are barely noticeable, other times are very noticeable. If I save the image and reload it back to imagick, I always get a different result in the end, which shouldn't be the case (except maybe for loss of precision due to 16 to 8 bit channels).

Another example here, even more trivial: https://stackoverflow.com/questions/69908182/
White pixels become colored for some reason.

**Comment by Danack** [comment]
Yeah....definitely looks like a bug. Using pseudo image:

```
$imagick = new Imagick();
$imagick->newPseudoImage(640, 480, "magick:logo");

$mask = new Imagick();
$mask->newImage($imagick->getImageWidth(), $imagick->getImageHeight(), "white");
$drawing = new ImagickDraw();
$drawing->setBorderColor('black');
$drawing->setFillColor('black');
$drawing->ellipse(
	$imagick->getImageWidth() / 2,
	$imagick->getImageHeight() / 2,
	$imagick->getImageWidth() / 3,
	$imagick->getImageWidth() / 4,
	0,
	360
);
$mask->drawImage($drawing);

//$mask->setImageFormat('png');
//$mask->writeImage(__DIR__ . "/mask.png");

$imagick->setImageMask($mask, Imagick::PIXELMASK_WRITE);
$imagick->brightnessContrastImage(-20, 10, Imagick::CHANNEL_ALL);
$imagick->unsharpMaskImage(1, 0.5, 1, 0);

$imagick->writeImage(__DIR__ . "/result1.png");
```

![result1](https://user-images.githubusercontent.com/1505719/141116892-65892cf5-a809-4eb0-b82c-002de1def9c8.png)

**Comment by Danack** [comment]
Reported upstream https://github.com/ImageMagick/ImageMagick/issues/4465

**Comment by Danack** [comment]
This appears to be fixed in ImageMagick-7.1.0-14. I'll ask CMB to upgrade to that version and rebuild, right after I tag the Imagick release.
